### PR TITLE
Fix Classic TBC Update Object parsing using wrong ObjectType enum

### PR DESCRIPTION
**Pull request by killerwife**

### CODE REVIEW

**Comment by ratkosrb** [comment]
I think it changed in 2.5.4.

**Comment by killerwife** [comment]
You are right, will fix.

**Comment by killerwife** [comment]
Fixed

File: WowPacketParserModule.V2_5_1_38707/Parsers/UpdateHandler.cs [hunk 1]
```diff
@@ -159,7 +159,11 @@ public static void HandleUpdateObject(Packet packet)
 
         private static void ReadCreateObjectBlock(Packet packet, CreateObject createObject, WowGuid guid, uint map, object index)
         {
-            ObjectType objType = ObjectTypeConverter.Convert(packet.ReadByteE<ObjectType801>("Object Type", index));
+            ObjectType objType;
+            if (ClientVersion.AddedInVersion(ClientVersionBuild.V2_5_4_42695))
```
Comment (mdX7) [inline WowPacketParserModule.V2_5_1_38707/Parsers/UpdateHandler.cs:163]: shouldnt this be RemovedInVersion according to brotalnias comment?

**Comment by ratkosrb** [comment]
They changed it to ObjectType801 in 2.5.4. Also since 1.14 uses the same handlers, you should check the 1.14 build here too and specify branch. And the HeirFlags can be read within the same if block, since it changed at the same time.

```
ObjectType objType;
if (ClientVersion.RemovedInVersion(ClientBranch.Classic, ClientVersionBuild.V1_14_3_42770) || 
    ClientVersion.RemovedInVersion(ClientBranch.TBC, ClientVersionBuild.V2_5_4_42695))
{
    objType = ObjectTypeConverter.Convert(packet.ReadByteE<ObjectTypeBCC>("Object Type", index));
    packet.ReadInt32("HeirFlags", index);
}
else
    objType = ObjectTypeConverter.Convert(packet.ReadByteE<ObjectType801>("Object Type", index));
```

File: WowPacketParserModule.V2_5_1_38707/Parsers/UpdateHandler.cs (hunk 1 above)
Comment (killerwife) [reply WowPacketParserModule.V2_5_1_38707/Parsers/UpdateHandler.cs]: I refactored it a bit to be more readable. Clarification - BCC flags PRE 2.5.4, 801 flags POST 2.5.4

**Pull request merged**